docs/src/metrics.rst

```diff
@@ -4,8 +4,64 @@
 Metrics operations
 ======================
 
-libuv provides a metrics API to track the amount of time the event loop has
-spent idle in the kernel's event provider.
+libuv provides a metrics API to track various internal operations of the event
+loop.
+
+
+Data types
+----------
+
+.. c:type:: uv_metrics_t
+
+    Metrics handle type.
+
+.. c:type:: uv_metrics_list_t
+
+    The struct of metrics that are passed to :c:type:`uv_metrics_cb`. These
+    metrics start being tracked internally when the :c:type:`uv_loop_t` is
+    initialized. So any call to the :c:type:`uv_metrics_cb` will have the most
+    recent metrics. This is useful when metrics only need to be sampled at
+    intervals.
+
+    ::
+
+        typedef struct {
+            uint64_t loop_count;
+            uint64_t events;
+            uint64_t events_waiting;
+        } uv_metrics_list_t;
+
+.. c:type:: void (*uv_metrics_cb)(uv_metrics_t* handle, uv_metrics_list_t* metrics)
+
+    Callback called just before the event loop enters the event provider.
+    Indicating the previous event loop iteration has completed.
+
```

Comment:
We should add `.. versionadded:: 1.45.0` in the documentation

Reply:
I'll make sure to add this just before it lands. Just in case there's another version bump before it does land.